### PR TITLE
Top commands / single Yoshi dependency for monorepos

### PR DESCRIPTION
**Pull request by ronami**
### 🔦 Summary

This PR changes how the `build` and `start` commands work for monorepos. Instead of spawning multiple `yoshi` commands, it builds the needed apps and libraries in a single flow.

I refactored `yoshi-config/paths`, `yoshi-config/loadConfig` and similar functions to accept `cwd`s other than `process.cwd()`.

I introduced the concept of `app`/`lib` to describe the information about packages in a monorepo. I used Lerna's APIs to get those packages.

### CODE REVIEW

File: packages/yoshi/config/webpack.config.js [hunk 1]
```diff
@@ -886,9 +886,9 @@ function createServerWebpackConfig({ isDebug = true, isHmr = false } = {}) {
         modulesDir: path.resolve(__dirname, '../node_modules'),
       }),
       // Treat monorepo (hoisted) dependencies as external
-      project.experimentalMonorepoSubProcess &&
+      rootApp.experimentalMonorepo &&
         nodeExternals({
-          modulesDir: path.resolve(MONOREPO_ROOT, 'node_modules'),
+          modulesDir: path.resolve(app.MONOREPO_ROOT, 'node_modules'),
```
Comment (ronami) [inline packages/yoshi/config/webpack.config.js]: rootApp.MONOREPO_ROOT

File: packages/yoshi/src/commands/build-app.js [hunk 2]
```diff
@@ -109,134 +58,29 @@ module.exports = async () => {
...
-  } else {
-    console.log(chalk.green('Compiled successfully.\n'));
-  }
+  const webpackStats = await runWebpack([
+    clientDebugConfig,
+    clientOptimizedConfig,
+    serverConfig,
+  ]);
 
   const clientOptimizedStats = webpackStats.stats[1];
```
Comment (ronami) [inline packages/yoshi/src/commands/build-app.js]: Destructure over stats[2] and use in other places in this file.

File: packages/yoshi/src/commands/build-monorepo.js [hunk 3]
```diff
@@ -1,64 +1,81 @@
...
+  const scopeFlags = libs.map(lib => `--scope=${lib.name}`);
 
-    assets.forEach(assetPath => {
-      fs.copyFileSync(
-        path.join(packageDirectory, assetPath),
-        path.join(packageDirectory, 'dist', assetPath),
-      );
+  try {
+    await execa.shell(`npx lerna exec ${scopeFlags.join(' ')} -- npx tsc`, {
+      stdio: 'inherit',
```
Comment (ronami) [inline packages/yoshi/src/commands/build-monorepo.js]: Use `pipe` instead.

File: packages/yoshi/src/commands/build-monorepo.js [hunk 4]
```diff
@@ -1,64 +1,81 @@
...
-      chalk.bold.red('Currently, Yoshi only support monorepos with one app.'),
-    );
+  // Calculate assets sizes
+  apps.forEach((app, index) => {
+    console.log(chalk.bold.underline(app.name));
     console.log();
 
-    process.exit(1);
-  }
+    const stats = [
```
Comment (ronami) [inline packages/yoshi/src/commands/build-monorepo.js]: Destructure with names.

File: packages/yoshi-helpers/monorepo.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+const execa = require('execa');
+const { partition } = require('lodash');
+const { getPaths } = require('yoshi-config/paths');
+const loadConfig = require('yoshi-config/loadConfig');
+
+const { stdout } = execa.shellSync('npx lerna list --all --json');
```
Comment (ronami) [inline packages/yoshi-helpers/monorepo.js]: Move to `yoshi-config`

File: packages/yoshi/bin/yoshi-cli.js [hunk 6]
```diff
@@ -49,10 +49,23 @@ if (config.experimentalMonorepo) {
     .action(() => runCLI('build-monorepo'));
 
   prog
-    .command('start')
+    .command('start [app]')
```
Comment (ronami) [inline packages/yoshi/bin/yoshi-cli.js]: use `<>` to enforce using it.

File: packages/yoshi/bin/yoshi-cli.js [hunk 7]
```diff
@@ -49,7 +49,7 @@ if (config.experimentalMonorepo) {
     .action(() => runCLI('build-monorepo'));
 
   prog
-    .command('start [app]')
+    .command('start <app>')
```
Comment (ranyitz) [inline packages/yoshi/bin/yoshi-cli.js]: ❤️

**Pull request closed without merging**